src/stdlib_stringlist_type.f90

```diff
@@ -0,0 +1,726 @@
+
+    type stringlist_type
+        private
+        type(string_type), dimension(:), allocatable :: stringarray
+    
+    contains
+        private
+
+        procedure, public :: clear                          =>  clear_list
+
+        procedure, public :: len                            =>  length_list
+
+        procedure         :: to_future_at_idxn              =>  convert_to_future_at_idxn
+
+        procedure         :: to_current_idxn                =>  convert_to_current_idxn
+
+        procedure         :: insert_at_char_idx             =>  insert_at_char_idx_wrap
+        procedure         :: insert_at_string_idx           =>  insert_at_string_idx_wrap
+        procedure         :: insert_at_stringlist_idx       =>  insert_at_stringlist_idx_wrap
+        procedure         :: insert_at_chararray_idx        =>  insert_at_chararray_idx_wrap
+        procedure         :: insert_at_stringarray_idx      =>  insert_at_stringarray_idx_wrap
+        generic, public   :: insert_at                      =>  insert_at_char_idx,         &
+                                                                insert_at_string_idx,       &
+                                                                insert_at_stringlist_idx,   &
+                                                                insert_at_chararray_idx,    &
+                                                                insert_at_stringarray_idx
+
+        procedure         :: insert_before_string_int       =>  insert_before_string_int_impl
+        procedure         :: insert_before_stringlist_int   =>  insert_before_stringlist_int_impl
+        procedure         :: insert_before_chararray_int    =>  insert_before_chararray_int_impl
+        procedure         :: insert_before_stringarray_int  =>  insert_before_stringarray_int_impl
+        generic           :: insert_before                  =>  insert_before_string_int,       &
+                                                                insert_before_stringlist_int,   &
+                                                                insert_before_chararray_int,    &
+                                                                insert_before_stringarray_int
+
+        procedure         :: get_string_idx         => get_string_idx_wrap
+        generic, public   :: get                    => get_string_idx
+
+    end type stringlist_type
+
+    !> Version: experimental
+    !>
+    !> Constructor for stringlist
+    !> Returns an instance of type stringlist_type
+    !> [Specifications](../page/specs/stdlib_stringlist_type.html#stringlist_type)
+    interface stringlist_type
+        module procedure new_stringlist
+        module procedure new_stringlist_carray
+    end interface
+
+    !> Version: experimental
+    !> 
+    !> Concatenates stringlist with the input entity
+    !> Returns a new stringlist
+    !> [Specifications](../page/specs/stdlib_stringlist_type.html#append-operator)
+    interface operator(//)
```

Comment:
Are these operators defined in the specs?

Reply:
Done